src/EnergyPlus/DaylightingManager.cc

```diff
@@ -6815,6 +6815,46 @@ void DayltgInteriorIllum(EnergyPlusData &state,
 
     } // ISWFLG /= 0 .AND. DaylIllum(1) > SETPNT(1)
 
+    // loop over windows to do luminance based control
+    int count = 0;
+    for (std::size_t igroup = 1; igroup <= thisDaylightControl.ShadeDeployOrderExtWins.size(); igroup++) {
+        std::vector<int> const &listOfExtWin = thisDaylightControl.ShadeDeployOrderExtWins[igroup - 1];
+        for (const auto IWin : listOfExtWin) {
+            ++count;
+            // need to map back to the original order of the "loop" to not change all the other data structures
+            int loop = thisDaylightControl.MapShdOrdToLoopNum(count);
+            if (loop > 0) {
+                ICtrl = state.dataSurface->Surface(IWin).activeWindowShadingControl;
+                if (!state.dataSurface->Surface(IWin).HasShadeControl) continue;
+                WinShadingType currentFlag = state.dataSurface->SurfWinShadingFlag(IWin);
+                WinShadingType ShType = state.dataSurface->WindowShadingControl(ICtrl).ShadingType;
+                WindowShadingControlType shCtrlType = state.dataSurface->WindowShadingControl(ICtrl).ShadingControlType;
+                if (((shCtrlType == WindowShadingControlType::HiLumin_HiSolar_OffMidNight) ||
+                     (shCtrlType == WindowShadingControlType::HiLumin_HiSolar_OffSunset) ||
+                     (shCtrlType == WindowShadingControlType::HiLumin_HiSolar_OffNextMorning)) &&
+                    ((currentFlag == WinShadingType::IntShadeConditionallyOff) || (currentFlag == WinShadingType::GlassConditionallyLightened) ||
+                     (currentFlag == WinShadingType::ExtShadeConditionallyOff) || (currentFlag == WinShadingType::IntBlindConditionallyOff) ||
+                     (currentFlag == WinShadingType::ExtBlindConditionallyOff) || (currentFlag == WinShadingType::BGShadeConditionallyOff) ||
+                     (currentFlag == WinShadingType::BGBlindConditionallyOff))) {
```

Comment:
Ummm ... so we don't have to do this anymore I don't think. Someone please check MSVC but g++ 4.2 does the `BITF` equivalent optimization. Even under `-O`!

Reply:
> Ummm ... so we don't have to do this anymore I don't think. Someone please check MSVC but g++ 4.2 does the `BITF` equivalent optimization. Even under `-O`!

Oh, ok, didn't know that.

Reply:
So, whatever g++ version we were on three years ago did not do this. I looked again a few weeks ago because of something Yujie asked me and was pleasantly surprised! Someone please check what the situation is with MSVC.

Reply:
> So, whatever g++ version we were on three years ago did not do this. I looked again a few weeks ago because of something Yujie asked me and was pleasantly surprised! Someone please check what the situation is with MSVC.

How would I check that in MSVC?